src/ORM/TableRegistry.php

```diff
@@ -55,212 +54,47 @@ class TableRegistry
-                list(, $table) = pluginSplit($options['className']);
-                $options['table'] = Inflector::underscore($table);
-            }
-            $options['className'] = 'Cake\ORM\Table';
-        }
-
-        if (isset(static::$_config[$alias])) {
-            $options += static::$_config[$alias];
+        if ($instance) {
+            static::$_instance = $instance;
         }
-        if (empty($options['connection'])) {
-            $connectionName = $options['className']::defaultConnectionName();
-            $options['connection'] = ConnectionManager::get($connectionName);
-        }
-
-        $options['registryAlias'] = $alias;
-        static::$_instances[$alias] = new $options['className']($options);
 
-        if ($options['className'] === 'Cake\ORM\Table') {
-            static::$_fallbacked[$alias] = static::$_instances[$alias];
+        if (!static::$_instance) {
+            static::$_instance = new static::$_defaultRegistryClass;
         }
 
-        return static::$_instances[$alias];
-    }
-
-    /**
-     * Check to see if an instance exists in the registry.
-     *
-     * @param string $alias The alias to check for.
-     * @return bool
-     */
-    public static function exists($alias)
-    {
-        return isset(static::$_instances[$alias]);
-    }
-
-    /**
-     * Set an instance.
-     *
-     * @param string $alias The alias to set.
-     * @param \Cake\ORM\Table $object The table to set.
-     * @return \Cake\ORM\Table
-     */
-    public static function set($alias, Table $object)
-    {
-        return static::$_instances[$alias] = $object;
-    }
-
-    /**
-     * Clears the registry of configuration and instances.
-     *
-     * @return void
-     */
-    public static function clear()
-    {
-        static::$_instances = [];
-        static::$_config = [];
-        static::$_fallbacked = [];
-    }
-
-    /**
-     * Returns the list of tables that were created by this registry that could
-     * not be instantiated from a specific subclass. This method is useful for
-     * debugging common mistakes when setting up associations or created new table
-     * classes.
-     *
-     * @return array
-     */
-    public static function genericInstances()
-    {
-        return static::$_fallbacked;
+        return static::$_instance;
     }
 
     /**
-     * Removes an instance from the registry.
+     * Proxy for static calls on a singleton.
      *
-     * @param string $alias The alias to remove.
-     * @return void
+     * @param string $name Method name.
+     * @param array $arguments Method arguments.
+     * @return mixed
      */
-    public static function remove($alias)
+    public static function __callStatic($name, $arguments)
     {
-        unset(
-            static::$_instances[$alias],
-            static::$_config[$alias],
-            static::$_fallbacked[$alias]
-        );
+        return call_user_func_array([static::instance(), $name], $arguments);
```

Comment:
I suggested using direct proxy for most common methods (see the `RegistryInterface`) and leave the rest to `__callStatic()` which is quite slow. I'll add them later...